### DISCUSSION TITLE
Invalid JSON Error on v9/projects

### DISCUSSION THREAD

**Discussion by mattbeiswenger** [vercel/vercel#8526] [Help] [answered]
I'm getting an error when trying to edit an environment variable using the Vercel API. My code currently looks like the following (secrets are obviously replaced):

```javascript
import fetch from 'node-fetch'

;(async function () {
  const response = await fetch(
    `https://api.vercel.com/v9/projects/<project_name>/env/<env_variable_id>`,
    {
      headers: {
        Authorization: 'Bearer <vercel_access_token>',
      },
      method: 'PATCH',
      body: {
        value: 'test',
      },
    }
  )
  const data = await response.json()
  console.log(data)
})()
```

The `console.log(data)` statement just outputs `{ error: { code: 400, message: 'Invalid JSON' } }`. I'm able to use the `curl` command from the [example docs](https://vercel.com/docs/rest-api#endpoints/projects/edit-an-environment-variable) just fine but for some reason I'm not having any luck using the fetch api.

**Comment by amyegan** [comment, marked as answer]
Hi @mattbeiswenger! Do you still get the 'Invalid JSON' error if you use [JSON.stringify()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) to format the `body` of the request?

**Reply by mattbeiswenger** [reply]
That worked! Thanks for the help!
